lib/rubocop/cop/lint/useless_assignment.rb

```diff
@@ -53,24 +52,27 @@ def after_leaving_scope(scope, _variable_table)
           scope.variables.each_value { |variable| check_for_unused_assignments(variable) }
         end
 
-        # rubocop:disable Metrics/AbcSize
+        # rubocop:disable Metrics/AbcSize, Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
         def check_for_unused_assignments(variable)
           return if variable.should_be_unused?
 
           variable.assignments.reverse_each do |assignment|
-            next if assignment.used? || part_of_ignored_node?(assignment.node)
+            assignment_node = assignment.node
+            next if assignment.used? || part_of_ignored_node?(assignment_node)
 
             message = message_for_useless_assignment(assignment)
             range = offense_range(assignment)
 
             add_offense(range, message: message) do |corrector|
-              autocorrect(corrector, assignment) unless sequential_assignment?(assignment.node)
+              if !sequential_assignment?(assignment_node) && !assignment_node.parent&.or_asgn_type?
```

Comment:
Might be good to add some comment here explaining what we're skipping for the auto-correction.

Reply:
Sure! I've added code comment for the logic.